### ISSUE TITLE
[Feature]: List files in a theme

### ISSUE THREAD

**Issue by alexantom** [Shopify/cli#2002]
### What type of project(s) will this request affect?

theme

### What type of change do you want to see?

New feature

### Overview

I'm currently working on shopify theme 2.0 to support multiple storefronts. As each store will end up creating different json templates from theme editor, it would be better to have a cli command to get list of all files and also based on subfolders.

Currently i'm maintaining list of files in shopifyignore. But during new storefront theme push i have to remove shopifyignore while  pushing new theme version to existing stores.

This will help me to write deploy scripts to ignore those json files and avoid overwriting changed json templates in theme editor

It would be also good to have a  cli command to get list of changed files from theme editor

So basically 2 request

1. Get list of files from current theme 
2. Get list of changed files

### Motivation

I'm currently working on shopify theme 2.0 to support multiple storefronts. As each store will end up creating different json templates from theme editor

**Comment by alexantom** [comment]
please keep it open
